client/my-sites/site-settings/form-base.js

```diff
@@ -62,6 +62,11 @@ module.exports = {
 
 	},
 
+	recordClickEventAndStop: function( recordObject, clickEvent ) {
+		this.recordEvent( recordObject );
+		clickEvent.preventDefault();
```

Comment:
@omarjackman : TLDR: `stopPropagation` doesent work.
I have tried that before, but I dont know what happens on Safari that even when I stop propagation, it still goes up the DOM tree.
This is the only solution that works. Maybe some problem with React synthetic events...